### DISCUSSION TITLE
Generate API doc from data object

### DISCUSSION THREAD

**Discussion by HugoHeneault** [spatie/laravel-data#109] [General]
Hi there, 

Quite new to this awesome plugin, loving it!

Is there a way to generate Swagger or swagger-like documentation from Data objects?

**Comment by freekmurze** [comment]
No, that isn't supported.

**Reply by gcaraciolo** [reply]
@freekmurze do you think this project should support this feature? I could work on a PR if you think it is worth. Otherwise, do you have some suggestion of a way to go for Api Doc?

**Reply by Nir-An** [reply]
Hi @gcaraciolo 

> do you think this project should support this feature? 

I don't think it should be supported either because Laravel-data 
is about objects in general and not necessarily about what the APIs are doing.

 (API can work without any object from Laravel-data and vice-versa
  objects from Laravel-data could be created without being sent to API)

> you have some suggestion of a way to go for Api Doc?

You could try   https://github.com/knuckleswtf/scribe
